aws/sra-test/integration-tests/aws-sdk-s3/benches/middleware_vs_orchestrator.rs

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#[macro_use]
+extern crate criterion;
+use aws_sdk_s3 as s3;
+use aws_smithy_client::erase::DynConnector;
+use aws_smithy_client::test_connection::infallible_connection_fn;
+use aws_smithy_runtime_api::type_erasure::TypedBox;
+use criterion::Criterion;
+use s3::operation::list_objects_v2::{ListObjectsV2Error, ListObjectsV2Input, ListObjectsV2Output};
+
+async fn middleware(client: &s3::Client) {
+    client
+        .list_objects_v2()
+        .bucket("test-bucket")
+        .prefix("prefix~")
+        .send()
+        .await
+        .expect("successful execution");
+}
+
+async fn orchestrator(connector: &DynConnector) {
+    // TODO(enableNewSmithyRuntime): benchmark with `send_v2` directly once it works
+    let runtime_plugins = aws_smithy_runtime_api::client::runtime_plugin::RuntimePlugins::new()
+        .with_client_plugin(orchestrator::ManualServiceRuntimePlugin(connector.clone()))
+        .with_operation_plugin(aws_sdk_s3::operation::list_objects_v2::ListObjectsV2::new())
+        .with_operation_plugin(orchestrator::ManualOperationRuntimePlugin);
+    let input = ListObjectsV2Input::builder()
+        .bucket("test-bucket")
+        .prefix("prefix~")
+        .build()
+        .unwrap();
+    let input = TypedBox::new(input).erase();
+    let output = aws_smithy_runtime::client::orchestrator::invoke(input, &runtime_plugins)
+        .await
+        .map_err(|err| {
+            err.map_service_error(|err| {
+                TypedBox::<ListObjectsV2Error>::assume_from(err)
+                    .expect("correct error type")
+                    .unwrap()
+            })
+        })
+        .unwrap();
+    TypedBox::<ListObjectsV2Output>::assume_from(output)
+        .expect("correct output type")
+        .unwrap();
+}
+
+fn test_connection() -> DynConnector {
+    infallible_connection_fn(|req| {
+        assert_eq!(
+            "https://test-bucket.s3.us-east-1.amazonaws.com/?list-type=2&prefix=prefix~",
+            req.uri().to_string()
+        );
+        assert!(req.headers().contains_key("authorization"));
```

Comment:
I think for the purposes of this specific benchmark, I want to keep these since I want to make sure we're actually comparing the same thing and that we're not comparing a broken orchestrator to a functioning middleware stack. Since this benchmark isn't trying to determine the actual overhead, but rather the difference in the overheads, I'm not worried about it.